dashboard/app/models/peer_review.rb

```diff
@@ -145,7 +145,19 @@ def self.get_peer_review_summaries(user, script)
         script.professional_learning_course? &&
         Plc::EnrollmentUnitAssignment.exists?(user: user, plc_course_unit: script.plc_course_unit)
 
-      PeerReview.where(reviewer: user, script: script).map(&:summarize)
+      PeerReview.where(reviewer: user, script: script).map(&:summarize).tap do |reviews|
+        if script.peer_reviews_to_complete &&
+            reviews.size < script.peer_reviews_to_complete &&
+            PeerReview.get_review_for_user(script, user)
+          reviews << {
+              status: 'not_started',
+              name: I18n.t('peer_review.review_new_submission'),
+              result: ActivityConstants::UNSUBMITTED_RESULT,
+              icon: '',
+              locked: false
+          }
```

Comment:
Summarize returns a list of summarized peer reviews for the purpose of rendering level progress dots. This is used on the client side to generate those dots, the same way they are generated for levels.

This part generates a line item that will show up as an empty progress dot that will redirect the user to the pull review path.

So you're right, it doesn't persist anything.